src/Mod/Fem/App/FemPostGroupExtension.cpp

```diff
@@ -0,0 +1,158 @@
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+#include "FemPostGroupExtension.h"
+#include <App/Document.h>
+
+using namespace Fem;
+
+EXTENSION_PROPERTY_SOURCE(Fem::FemPostGroupExtension, App::GroupExtension);
+
+const char* FemPostGroupExtension::ModeEnums[] = {"Serial", "Parallel", nullptr};
+
+FemPostGroupExtension::FemPostGroupExtension()
+    : App::GroupExtension()
+{
+
+    initExtensionType(Fem::FemPostGroupExtension::getExtensionClassTypeId());
+
+    EXTENSION_ADD_PROPERTY_TYPE(
+        Mode,
+        (long(0)),
+        "Pipeline",
+        App::Prop_None,
+        "Selects the pipeline data transition mode.\n"
+        "In serial, every filter gets the output of the previous one as input.\n"
+        "In parallel, every filter gets the pipeline source as input.\n");
+
+    Mode.setEnums(ModeEnums);
+}
+
+FemPostGroupExtension::~FemPostGroupExtension()
+{}
+
+
+void FemPostGroupExtension::initExtension(App::ExtensionContainer* obj)
+{
+
+    if (!obj->isDerivedFrom(Fem::FemPostObject::getClassTypeId())) {
+        throw Base::RuntimeError("FemPostGroupExtension can only be applied to FemPostObject");
+    }
+
+    App::GroupExtension::initExtension(obj);
+}
+
+void FemPostGroupExtension::extensionOnChanged(const App::Property* p)
+{
+    if (p == &Group) {
+        if (!m_blockChange) {
+            // sort the group, so that non filter objects are always on top (in case any object
+            // using this extension allows those)
+            auto objs = Group.getValues();
+            std::sort(objs.begin(),
+                      objs.end(),
+                      [](const App::DocumentObject* lhs, const App::DocumentObject* rhs) {
+                          int l = lhs->isDerivedFrom(FemPostFilter::getClassTypeId()) ? 0 : 1;
+                          int r = rhs->isDerivedFrom(FemPostFilter::getClassTypeId()) ? 0 : 1;
+                          return r < l;
+                      });
+            m_blockChange = true;
+            Group.setValue(objs);
+            m_blockChange = false;
+        }
+    }
+    GroupExtension::extensionOnChanged(p);
+}
+
+std::vector<Fem::FemPostFilter*> FemPostGroupExtension::getFilter()
+{
+    // collect all other items that are not filters
+    std::vector<Fem::FemPostFilter*> filters;
+    for (auto obj : Group.getValues()) {
+        if (obj->isDerivedFrom(Fem::FemPostFilter::getClassTypeId())) {
+            filters.push_back(static_cast<FemPostFilter*>(obj));
+        }
+    }
+    return filters;
+}
+
+App::DocumentObject* FemPostGroupExtension::getGroupOfObject(const App::DocumentObject* obj)
+{
+    for (auto o : obj->getInList()) {
+        if (o->hasExtension(FemPostGroupExtension::getExtensionClassTypeId(), false)) {
```

Comment:
```suggestion
    for (auto o& : obj->getInList()) {
        if (o->hasExtension<FemPostGroupExtension>(false)) {
```
Idk if the angle brackets here are possible

Reply:
No, that is not implemented.